awx/main/models/inventory.py

```diff
@@ -2036,9 +2036,25 @@ def inventory_as_dict(self, inventory_update, private_data_dir):
         for key, loc in old_filterables:
             value = source_vars.get(key, None)
             if value and isinstance(value, str):
-                user_filters.append('{} not in {}'.format(
-                    loc, value.split(',')
-                ))
+                # tags can be list of key:value pairs
+                #  e.g. 'Creator:jmarshall, peanutbutter:jelly'
+                # or tags can be a list of keys
+                #  e.g. 'Creator, peanutbutter'
+                if key == "tags":
+                    # grab each key value pair
+                    for kvpair in value.split(','):
+                        # split into key and value
+                        kv = kvpair.split(':')
+                        # filter out any host that does not have key
+                        # in their tags.keys() variable
+                        user_filters.append('"{}" not in tags.keys()'.format(kv[0].strip()))
+                        # if a value is provided, check that the key:value pair matches
+                        if len(kv) > 1:
+                            user_filters.append('tags["{}"] not in ["{}"]'.format(kv[0].strip(), kv[1].strip()))
```

Comment:
This `in <a list>` syntax doesn't make sense to me if unless `kv[1]` is split on some delimiter to allow multiple values. If not (which is fine), then I'd rather see this read as `tags["{}"] != "{}"`, as long as that works

Reply:
yeah this makes sense since we're assuming the value (in the key:value pair) is a single string, not a list.